utils/techutils/techutils.go

```diff
@@ -507,3 +532,68 @@ func GetAllTechnologiesList() (technologies []Technology) {
 	}
 	return
 }
+
+// SplitComponentId splits a Xray component ID to the component name, version and package type.
+// In case componentId doesn't contain a version, the returned version will be an empty string.
+// In case componentId's format is invalid, it will be returned as the component name
+// and empty strings will be returned instead of the version and the package type.
+// Examples:
+//  1. componentId: "gav://antparent:ant:1.6.5"
+//     Returned values:
+//     Component name: "antparent:ant"
+//     Component version: "1.6.5"
+//     Package type: "Maven"
+//  2. componentId: "generic://sha256:244fd47e07d1004f0aed9c156aa09083c82bf8944eceb67c946ff7430510a77b/foo.jar"
+//     Returned values:
+//     Component name: "foo.jar"
+//     Component version: ""
+//     Package type: "Generic"
+//  3. componentId: "invalid-comp-id"
+//     Returned values:
+//     Component name: "invalid-comp-id"
+//     Component version: ""
+//     Package type: ""
+func SplitComponentId(componentId string) (string, string, string) {
```

Comment:
Lets add a test for this function since parsing can be very easily broken and it is very hard to understand that later without a test

Reply:
this is the method we always used. I just moved it to a different place.